lib/growthbook/context.rb

```diff
@@ -88,35 +92,40 @@ def forced_features=(forced_features)
 
     def eval_feature(key)
       # Forced in the context
-      return get_feature_result(@forced_features[key.to_s], 'override') if @forced_features.key?(key.to_s)
+      return get_feature_result(@forced_features[key.to_s], 'override', nil, nil) if @forced_features.key?(key.to_s)
 
       # Return if we can't find the feature definition
       feature = get_feature(key)
-      return get_feature_result(nil, 'unknownFeature') unless feature
+      return get_feature_result(nil, 'unknownFeature', nil, nil) unless feature
 
       feature.rules.each do |rule|
         # Targeting condition
-        next if rule.condition && !condition_passes(rule.condition)
+        next if rule.condition && !condition_passes?(rule.condition)
 
         # If there are filters for who is included (e.g. namespaces)
-        next if rule.filters && filtered_out?(rule.filters)
+        next if rule.filters && filtered_out?(rule.filters || [])
 
         # If this is a percentage rollout, skip if not included
         if rule.force?
           seed = rule.seed || key
           hash_attribute = rule.hash_attribute || 'id'
           included_in_rollout = included_in_rollout?(
-            seed: seed, hash_attribute: hash_attribute, range: rule.range,
-            coverage: rule.coverage, hash_version: rule.hash_version
+            seed: seed.to_s,
+            hash_attribute: hash_attribute,
+            range: rule.range,
+            coverage: rule.coverage,
+            hash_version: rule.hash_version
           )
           next unless included_in_rollout
 
-          return get_feature_result(rule.force, 'force')
+          return get_feature_result(rule.force, 'force', nil, nil)
         end
         # Experiment rule
         next unless rule.experiment?
 
         exp = rule.to_experiment(key)
+        next if exp.nil? # TODO: Is this correct? exp is nillable
```

Comment:
@jdorn is this the right logic here? Tests seem to pass with this but not sure if a nil experiment key is covered in tests.